assets/js/components/ReportError.js

```diff
@@ -51,43 +52,81 @@ export default function ReportError( { moduleSlug, error } ) {
 
 	const dispatch = useDispatch();
 
-	let title = sprintf(
-		/* translators: %s: module name */
-		__( 'Data error in %s', 'google-site-kit' ),
-		module?.name
-	);
-	let message = error.message;
+	const errors = Array.isArray( error ) ? error : [ error ];
+	const uniqueErrors = uniqueBy( errors, ( e ) => e.code && e.message );
 
-	if ( isInsufficientPermissionsError( error ) ) {
+	let title;
+	if ( uniqueErrors.length > 1 ) {
+		title = sprintf(
+			/* translators: %s: module name */
+			__( 'Data errors in %s', 'google-site-kit' ),
+			module?.name
+		);
+	} else {
 		title = sprintf(
 			/* translators: %s: module name */
-			__( 'Insufficient permissions in %s', 'google-site-kit' ),
+			__( 'Data error in %s', 'google-site-kit' ),
 			module?.name
 		);
-		message = getInsufficientPermissionsErrorDescription( message, module );
 	}
 
-	const reconnectURL = error?.data?.reconnectURL;
-	const description = reconnectURL ? (
-		<ErrorText message={ message } reconnectURL={ reconnectURL } />
-	) : (
-		purify.sanitize( message, { ALLOWED_TAGS: [] } )
+	const getMessage = ( e ) => {
+		const message = e.message;
+
+		if ( isInsufficientPermissionsError( e ) ) {
+			title = sprintf(
+				/* translators: %s: module name */
+				__( 'Insufficient permissions in %s', 'google-site-kit' ),
+				module?.name
+			);
+			return getInsufficientPermissionsErrorDescription(
+				message,
+				module
+			);
+		}
+
+		return message;
+	};
+
+	const description = (
+		<Fragment>
+			{ uniqueErrors.map( ( e ) => {
+				const message = getMessage( e );
+				const reconnectURL = error?.data?.reconnectURL;
+				return reconnectURL ? (
+					<ErrorText
+						key={ e.message }
```

Comment:
I don't think `message` is a good choice of key here, as there's a chance that multiple errors may occur with the same `message` but a different `code` or `reconnectURL`. Bit of an edge case, but the key should either reflect the set of values that makes an error unique (i.e. `code`, `message` and `reconnectURL`) - or simply use `index`...